src/slowmokit/methods/neighbors/bernoulli_nb/bernoulli_nb.cpp

```diff
@@ -6,57 +6,61 @@
 #include "bernoulli_nb.hpp"
 
 template<class T>
-    double prior_prob(std::vector<T> y_train,int label){   // Prior-Probability P(y)
+    double prior_prob(std::vector<T> yTrain,int label){   // Prior-Probability P(y)
         int sum=0;
-        for(int i=0;i<y_train.size();i++){
-            if(y_train[i]==label){
+        for(int i=0;i<yTrain.size();i++){
+            if(yTrain[i]==label){
                 sum += 1;
             }
         }
-        return sum/double(y_train.size());
+        return sum/double(yTrain.size());
     }
 
 template<class T>
-    double conditional_prob(std::vector<std::vector<T>> x_train,std::vector<T> y_train,int feature_col,int feature_val,int label){ // Conditional Probability
+    double conditional_prob(std::vector<std::vector<T>> xTrain,std::vector<T> yTrain,int featureCol,int featureVal,int label){ // Conditional Probability
     // P(x=f1 / y=class) 
         int denominator=0,numerator=0;
-        for(int i=0;i<y_train.size();i++){
-            if(y_train[i]==label){
+        for(int i=0;i<yTrain.size();i++){
+            if(yTrain[i]==label){
                 denominator++;
-                if(x_train[i][feature_col]==feature_val){
+                if(xTrain[i][featureCol]==featureVal){
                     numerator++;
                 }
             }
         }
-        return numerator/double(denominator);
+        if(denominator!=0){
+            return numerator/double(denominator);
+        }
+        return numerator;
+        
     }
 
 template<class T>
-    int predict(std::vector<std::vector<T>> x_train,std::vector<T> y_train,std::vector<T> x_test){
-        int n_features = x_train[0].size();
+    int fit(std::vector<std::vector<T>> xTrain,std::vector<T> yTrain,std::vector<T> xTest){
```

Comment:
You need to make 2 different functions for fit and predict. Also make a function fit_predict...basically the current fit function